### ISSUE TITLE
set `$socket_group_default = undef` in params.pp to let it be overridable in hiera

### ISSUE THREAD

**Issue by androidkh** [puppetlabs/puppetlabs-docker#369]
Hi all,

currently there is a parameter $socket_group_default that is used to define $socket_group that in its turn is used in template docker.systemd.erb:

```
<% if @socket_group %> -G <%= @socket_group %><% end -%>
<% if @dns %><% @dns_array.each do |address| %> --dns <%= address %><% end %><% end -%>
```

By default on centos7 it resulted in adding "-G docker" to OPTIONS.
I have existing infrastructure that has no -G specified at all (and thus it's a root by default).
I found no way to override it in hiera. Even when I specified empty string -  <% if @socket_group %> was true and I just received `... -G --dns...`
When I tried to specify undef for this param - then it took default value 'docker' from params.pp

Is it possible to modify the code to be able to set $socket_group to undef?
I found the only way - not to assign any value in param.pp. In this case socket_group can by set in hiera. Any other ways are welcome as well.
Thanks, 
docker module version: 3.0.0

**Comment by mojibake-umd** [comment]
This is a dup of #321 in my opinion.
So current situation, broke.

**Comment by esalberg** [comment]
Following up on this - we used to have the same issue, but with the updated socket management for RHEL in Docker 18.09.1, setting socket_group to undef wouldn't work anyway.  We need to set it explicitly to root and add a /etc/systemd/system/docker.socket.d/socket-overrides.conf file to override the new default in /usr/lib/systemd/system/docker.socket (SocketGroup=docker).  It does mean that you'll have "-G root" in your OPTIONS line.

Which OS are you using?  I can only test with RHEL, but the ability is there to specify an override template for other types of Linux as well (disabled by default across the board).  I'll be putting in a PR shortly and will link.

**Comment by androidkh** [comment]
Idea was to allow "not to specify" -G option at all  as docker can use default value when no -G flag.

**Comment by esalberg** [comment]
In my PR, that is done by setting:

docker::socket_group: false

Since all the templates have <% if @socket_group %> wrappers around the socket_group related entries (including the -G option), it works fine in my testing.

**Comment by davejrt** [comment]
closed by #441
